Add Speakers section

dofxo opened the pull request:
This PR has been created to review the merge of the speakers section branch into the dev branch.

Points to be addressed:

Hardcoded Styles:
Some styles are currently hardcoded. Global styles, reusable classes, and standardized fonts should be created to address this issue.

Speakers Data:
The current speakers' data is placeholder content and needs to be replaced with the actual data.

RTL Direction:
The initiated PR lacks RTL support in the HTML direction, resulting in the output being displayed in LTR. This should be updated to align with RTL requirements where applicable.

dofxo committed:
feat: add speakers section's wrapper with its title

in this commit i've added speaker section wrapper with its styles along
with  the title of it.
as mentioned in the code, hard coded sizes will be replaced with
tailwind custom configs (color, spacings at etc.)

dofxo committed:
feat: add speakers data

in this commit i've added speakers data (not real data) to be able to
use them for the speakers sections
all the datas will be removed/replaced with the actual speakers data

dofxo committed:
feat: add sample company-logo and profile-picture for the speaker section

these images are sample images for the speakers and will be repalced
with actual data

dofxo committed:
feat: add speaker component

note: the current output of the card needs a rtl direction so the output
will become just like the design (it currently doesnt have the direction
as it is something to be placed in the layout)

dofxo committed:
modify: render speakers using speakers data

AmirHosseinKarimi commented:
- به نظر فونت‌ها بر اساس طرح تنظیم نشدن
- تنظیم rtl بودن صفحه انجام نشده
- حاشیه (Border) مربوط به اسم شرکت هر سخنران تنظیم نشده
<details><summary>Browser metadata</summary>





```
Path:      /
Browser:   Firefox 134.0 on Windows 10
Viewport:  1987 x 1279 @1x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-8--fc-nextjs.netlify.app%2F&os=Windows&os_version=10&browser=Firefox&browser_version=134.0&resolution=1987x1279&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-8--fc-nextjs.netlify.app/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=6799d98b23becd000887b7b6&commentID=679ca59385650992859a8482&resolution=resolved)

AmirHosseinKarimi commented on public/images/speakers/profile-picture/pooria-baba-ali.svg:
تصویر سخنران نیازی نیست که svg باشه. JPG یا WebP مناسب‌تره

AmirHosseinKarimi commented on src/components/sections/types.ts:
تایپ‌ها باید کنار هر کامپوننت بصورت `Speakers.d.ts` قرار بگیرن
مگر اینکه تایپ کلی یا نامرتبط با کامپوننت فعلی باشن که باید تو مسیر `src/types/something.d.ts` قرار بگیرن

AmirHosseinKarimi commented on src/data/speakers.ts:
کانفیگ‌ها و داده‌های قابل تغییر رو تو مسیر `src/configs/*` قرار بدیم بهتره به نظرم.

AmirHosseinKarimi reviewed (changes requested):
خسته نباشی محمد جان.
لطفا مواردی که ذکر کردم رو اعمال کن تا بتونم PR رو مرج کنم.
ممنون

dofxo commented:
ممنونم بابت نکاتی که گفتید🙏🏻

برای تغییراتی که گفتید مربوط فونت و rtl 
چون گفتم یه چیز گلوباله و به کانفلیکت بخوره شاید بعدا انجامشون ندادم 
پس مشکلی نداره خودم توی layout اعمالشون کنم؟

AmirHosseinKarimi commented:
> ممنونم بابت نکاتی که گفتید🙏🏻
> 
> برای تغییراتی که گفتید مربوط فونت و rtl چون گفتم یه چیز گلوباله و به کانفلیکت بخوره شاید بعدا انجامشون ندادم پس مشکلی نداره خودم توی layout اعمالشون کنم؟

سلامت باشی. نه کاملا اوکیه. هر جایی که نیاز به اعمال کانفیگ هست میتونیم تو همون برنچ اضافه‌اش کنیم.

dofxo committed:
feat: add rtl direction to main layout

dofxo committed:
refactor: change data directory name to configs

dofxo committed:
refactor: change pictures format

in this commit i've changed speakers img format to webp and company logo
to png format

dofxo committed:
refactor: change image styles in speakers section

in this commit i've changed some images styles to sync the new img
format changes
also changed some speaker component styles for better output

dofxo committed:
refactor: use tailwind colors instead of hard coded color

in this commit i've replaced hard coded colors for background color with
tailwind color classes

dofxo committed:
feat: add borderstyles to speakers company logo

dofxo committed:
refactor: change company logo width

dofxo committed:
refactor: add a wrapper for company logo for better aligment

dofxo committed:
remove: remove dark prefrence theme

in this commit i've removed the dark theme prefrence from global css as
it was not needed.

dofxo committed:
feat: add custom spacing to tailwind config

in this commit i've added custom spacing based on design variables

dofxo committed:
refactor: replace hard coded styles with tailwind custom classes

dofxo committed:
refactor: change folder structure

in this commit i've moved speaker section files from src/components/ to
app/components

dofxo committed:
refactor: rename speakers file type to d.ts

dofxo commented:
سلام، وقتتون بخیر
در رابطه با تغییراتی که قرار بود برای فایل تایپ‌ها انجام بدیم و فایلی که تایپ‌های مربوط به کانفیگ رو داره، به `Speakers.d.ts` تغییر نام بدیم، به یک مشکل برخوردم.

چنتا داکیومنت خوندم و گشتم، همه می‌گفتن ‍TypeScript اولویت‌بندی خاصی داره و موقع ایمپورت، اول دنبال فایل‌های `.tsx` یا `.ts` می‌گرده و بعدش `d.ts` ها. برای همین، وقتی از مسیر `./Speakers` تایپ‌ها رو ایمپورت می‌کردم، TypeScript سعی می‌کرد از کامپوننت `Speakers` تایپ‌ها رو بخونه که اصلاً وجود نداشت.

با تغییر نام فایل به `Speakers.types.d.ts`، مشکل حل شد (یا اینکه میشد مسیر فایل تایپ رو عوض کنم).
می‌خواستم ببینم آیا راه‌حل درست‌تر و اصولی‌تری برای این کار وجود داره که اونو انجام بدم؟

dofxo committed:
feat: add Dana font to project and remove the default nextjs font

dofxo committed:
refactor: change sample speaker role text

dofxo committed:
refactor: change speakers gap

in this commit i've changed the speakers gap to make the texts more
readable because speaker cards were too close to each other

dofxo committed:
feat: add muted-foreground color to tailwind colors

dofxo committed:
refactor: replace hard coded color with tailwind variable

dofxo commented:
تغییرات گفته شده اعمال شد

@AmirHosseinKarimi

dofxo committed:
feat: add speakers data

dofxo committed:
feat: add speakers company website links

dofxo committed:
refactor: change speaker picture

change sajjad maneshi profile-picture

AmirHosseinKarimi commented on src/app/page.tsx:
```diff
@@ -1,4 +1,4 @@
-import Speakers from "@/components/sections/speakers/Speakers";
+import Speakers from "./components/speakers/Speakers";
```
ایمپورت‌‌ها باید طبق کانفیگ انجام شده روی پروژه بصورت alias رو absolute باشه
`@/components/...`

AmirHosseinKarimi commented on public/fonts/dana/dana.css:
روش ایمپورت فونت مناسب نیست.
باید طبق داکیومنت از فونت‌ها استفاده کنیم
https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

AmirHosseinKarimi commented on src/app/components/speakers/Speaker.tsx:
```diff
@@ -0,0 +1,60 @@
+import { speakerInfoType } from "./Speakers.types";
```
در مورد ایمپورت تایپ‌ها میتونی اسم فایل رو بصورت کامل همراه با پسوند بیاری
`import { ISpeaker } from "@/Components/Speakers/Speaker.d.ts"`

dofxo replied:
این مورد رو هم که امتحان کردم نشد باز متاسفانه
میگه همچین آیتمی برای اکسپورت کردن وجود نداره 
مگر اینکه اسم فایلشو متفاوت بزارم یا مسیر این دو فایل متفاوت باشه، اون موقع میشه

dofxo committed:
refactor: change import path for Speakers section

dofxo committed:
refactor: change the way of using font to nextjs localFont

dofxo committed:
refactor: rename speaker interface name

change interface from speakerInfoType to ISpeaker

AmirHosseinKarimi committed:
refactor: rename speakerInfoType to ISpeakerInfo

AmirHosseinKarimi reviewed (changes requested):
ایمپورت تایپ رو فیکس کردم
اما فونت همچنان با CSS داره ایمپورت میشه

dofxo committed:
Merge remote-tracking branch 'origin/speakers' into speakers

dofxo commented:
> ایمپورت تایپ رو فیکس کردم اما فونت همچنان با CSS داره ایمپورت میشه

اونارو هم پوش کردم

dofxo committed:
refactor: change import type for ISpeaker in configs

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: create .all-contributorsrc [skip ci]

AmirHosseinKarimi committed:
Merge pull request #13 from frontChapter/all-contributors/add-fulcain

docs: add fulcain as a contributor for code

AmirHosseinKarimi committed:
Merge branch 'dev' into feat/improve-ci-cd-setup

AmirHosseinKarimi committed:
Merge pull request #10 from frontChapter/feat/improve-ci-cd-setup

ci: setup all-contributor bot

dofxo committed:
Merge branch 'dev' into speakers

dofxo committed:
refactor: change color variable

dofxo committed:
update: update package-lock

dofxo committed:
Merge branch 'dev' into speakers

dofxo committed:
fix: fix tailwind config issue

this issues has been made from resolving conflicts from dev branch

dofxo committed:
refactor: add dark color scheme support for texts

dofxo committed:
fix: correct dark border color for tailwind class

dofxo committed:
refactor: change text-card-foreground to text-muted-foreground

Bravemobin committed:
feat: Add Sponsors section

Bravemobin committed:
feat: Add sponsors section to the website

AmirHosseinKarimi committed:
Merge pull request #17 from frontChapter/sponsers

Add Sponsers Section

allcontributors[bot] committed:
docs: update README.md

allcontributors[bot] committed:
docs: update .all-contributorsrc

AmirHosseinKarimi committed:
Merge pull request #20 from frontChapter/all-contributors/add-Bravemobin

docs: add Bravemobin as a contributor for code

dofxo committed:
Merge branch 'dev' into speakers

AmirHosseinKarimi committed:
Merge branch 'dev' into speakers

AmirHosseinKarimi committed:
fix: rename Sponsers to Sponsors

AmirHosseinKarimi committed:
fix: rename Sponsers to Sponsors

AmirHosseinKarimi commented:
@all-contributors please add @dofxo  as a contributor to their code.
